.github/workflows/ci-checks.yml

```diff
@@ -0,0 +1,46 @@
+# This workflow runs on pull requests when they are opened, synchronized, or reopened.
+# It performs the following actions:
+# 1. Removes the 'ci-passed' label when a pull request is synchronized or reopened, 
+#    indicating that changes have been pushed, and CI needs to rerun.
+# 2. Polls every 5 minutes to check if all CI checks have passed.
+# 3. Adds the 'ci-passed' label to the pull request if all CI checks pass successfully.
+
+name: Check CI Completion
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  checks: read  
+  pull-requests: write
+
+jobs:
+  remove_ci_passed_label:
```

Comment:
similar to below, perhaps we should rename this to say `reset` instead of `remove`